x-pack/plugins/security_solution/public/management/links.ts

```diff
@@ -232,23 +232,32 @@ export const getManagementFilteredLinks = async (
 ): Promise<LinkItem> => {
   try {
     const currentUserResponse = await plugins.security.authc.getCurrentUser();
-    const privileges = calculateEndpointAuthz(
-      licenseService,
-      plugins.fleet?.authz,
-      currentUserResponse.roles
-    );
-    if (!privileges.canAccessEndpointManagement) {
+    const { canAccessEndpointManagement, canIsolateHost, canAccessResponseActionsHistory } =
+      calculateEndpointAuthz(licenseService, plugins.fleet?.authz, currentUserResponse.roles);
+
+    if (!canAccessEndpointManagement) {
       return excludeLinks([
         SecurityPageName.hostIsolationExceptions,
         SecurityPageName.responseActionsHistory,
       ]);
     }
-    if (!privileges.canIsolateHost || !privileges.canAccessResponseActionsHistory) {
-      const hostIsolationExceptionsApiClientInstance = HostIsolationExceptionsApiClient.getInstance(
-        core.http
-      );
-      const summaryResponse = await hostIsolationExceptionsApiClientInstance.summary();
+    const hostIsolationExceptionsApiClientInstance = HostIsolationExceptionsApiClient.getInstance(
+      core.http
+    );
+    const summaryResponse = await hostIsolationExceptionsApiClientInstance.summary();
```

Comment:
Can we do this inside of the if conditions so we don't do extra API calls if no needed? As it has to be done inside two if conditions you can move this inside a function and call it whenever is needed.

Reply:
This is computed only once essentially, cause the if conditions have return statements. If however, the conditions were not returning then there is a chance that the API request is made more than once. The only reason I see here to abstract it is to make it more readable.

Reply:
ecabceff24a8df701f563c024b122f7f12f677ea